File: src/llm_usage/privacy.py
from __future__ import annotations

from .feishu_schema import REQUIRED_FEISHU_FIELDS, field_names
from .models import AggregateRecord


UPLOAD_FIELD_ORDER = tuple(field_names(REQUIRED_FEISHU_FIELDS))
UPLOAD_FIELDS = set(UPLOAD_FIELD_ORDER)


def to_feishu_fields(row: AggregateRecord) -> dict[str, object]:
    raw = row.__dict__.copy()
    return {k: raw[k] for k in UPLOAD_FIELD_ORDER}

File: node/src/core/privacy.js
import { REQUIRED_FEISHU_FIELDS, fieldNames } from "../runtime/feishu-schema.js";

export const UPLOAD_FIELDS = new Set(fieldNames(REQUIRED_FEISHU_FIELDS));

export function toFeishuFields(row) {
  return Object.fromEntries(Object.entries(row).filter(([key]) => UPLOAD_FIELDS.has(key)));
}
